lib/googleauth/compute_engine.rb

```diff
@@ -83,11 +85,20 @@ def fetch_access_token options = {}
         c = options[:connection] || Faraday.default_connection
         retry_with_error do
           headers = { "Metadata-Flavor" => "Google" }
-          resp = c.get COMPUTE_AUTH_TOKEN_URI, nil, headers
+          resp =
+            if target_audience
+              c.get COMPUTE_ID_TOKEN_URI, { "audience" => target_audience }, headers
```

Comment:
So due to a GCE issue (they don't include the `email` field in the returned ID Token JWT, which unfortunately IAP requires, so making calls to IAP from GCE will fail with the implementation you have here), we need to include `format=full` in order to have the `email` field returned with the ID token.

The solution is just to add a querystring parameter `format=full` here, which I believe will just look like:

```rb
 c.get COMPUTE_ID_TOKEN_URI, { "audience" => target_audience, "format" => "full" }, headers
```

Reply:
done